hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -468,4 +478,83 @@ public void testUUIDinTableProperties() throws Exception {
+      // check whether parameters are in expected state
+      Map<String, String> parameters = hmsTable.getParameters();
+      Assert.assertEquals("0", parameters.get(TableProperties.SNAPSHOT_COUNT));
+      Assert.assertNull(parameters.get(TableProperties.CURRENT_SNAPSHOT_SUMMARY));
+      Assert.assertNull(parameters.get(TableProperties.CURRENT_SNAPSHOT_ID));
+      Assert.assertNull(parameters.get(TableProperties.CURRENT_SNAPSHOT_TIMESTAMP));
+
+      // create a snapshot
+      Table icebergTable = catalog.loadTable(tableIdentifier);
+      String fileName = UUID.randomUUID().toString();
+      DataFile file = DataFiles.builder(icebergTable.spec())
+          .withPath(FileFormat.PARQUET.addExtension(fileName))
+          .withRecordCount(2)
+          .withFileSizeInBytes(0)
+          .build();
+      icebergTable.newFastAppend().appendFile(file).commit();
+
+      // check whether parameters are in expected state
+      hmsTable = metastoreClient.getTable(tableIdentifier.namespace().level(0), tableName);
+      parameters = hmsTable.getParameters();
+      Assert.assertEquals("1", parameters.get(TableProperties.SNAPSHOT_COUNT));
+      String summary = JsonUtil.mapper().writeValueAsString(icebergTable.currentSnapshot().summary());
+      Assert.assertEquals(summary, parameters.get(TableProperties.CURRENT_SNAPSHOT_SUMMARY));
+      long snapshotId = icebergTable.currentSnapshot().snapshotId();
+      Assert.assertEquals(String.valueOf(snapshotId), parameters.get(TableProperties.CURRENT_SNAPSHOT_ID));
+      Assert.assertEquals(String.valueOf(icebergTable.currentSnapshot().timestampMillis()),
+          parameters.get(TableProperties.CURRENT_SNAPSHOT_TIMESTAMP));
+
+    } finally {
+      catalog.dropTable(tableIdentifier);
+    }
+  }
+
+  @Test
+  public void testSetSnapshotSummary() throws Exception {
+    Configuration conf = new Configuration();
+    conf.set("iceberg.hive.table.parameter.size.max", "4000");
+    HiveTableOperations spyOps = spy(new HiveTableOperations(conf, null, null, catalog.name(), DB_NAME, "tbl"));
+    Snapshot snapshot = mock(Snapshot.class);
+    Map<String, String> summary = Maps.newHashMap();
+    when(snapshot.summary()).thenReturn(summary);
+
+    // create a snapshot summary whose json string size is less than the limit
+    for (int i = 0; i < 100; i++) {
+      summary.put(String.valueOf(i), "value");
+    }
+    Assert.assertTrue(JsonUtil.mapper().writeValueAsString(summary).length() < 4000);
+    Map<String, String> parameter = Maps.newHashMap();
+    spyOps.setSnapshotSummary(parameter, snapshot);
+    Assert.assertEquals("The snapshot summary must be in parameters", 1, parameter.size());
+
+    // create a snapshot summary whose json string size exceeds the limit
+    for (int i = 0; i < 1000; i++) {
+      summary.put(String.valueOf(i), "value");
+    }
+    long summarySize = JsonUtil.mapper().writeValueAsString(summary).length();
+    Assert.assertTrue(summarySize > 4000 && summarySize < 32672);
```

Comment:
Nit: i think no need for the upper limit check, right?  We are not really saving in Hive.

Reply:
This is to make sure the new limit takes affect.
```
    conf.set("iceberg.hive.table.parameter.size.max", "4000");
```

Reply:
Add a comment in the code

Reply:
Hm, I meant, can't we just do 

```Assert.assertTrue("Summary size should be greater than limit", summarySize > 4000).  ```

To me the test is checking whether you can persist something beyond 4000 chars right?  The fact that it's below or above 32627 chars should not trigger now that we changed limit to 4000 right?

Reply:
The main purpose is to test whether we save the summary in HMS parameters when the size exceeds the limit. Besides, I also want to test if the limit has changed from 32627 to 4000. That's why I check both.
```
 Assert.assertTrue(summarySize > 4000 && summarySize < 32672);
```

Reply:
OK got it.